lambda_code/src/fhir_controller.py

```diff
@@ -99,23 +105,80 @@ def delete_immunization(self, aws_event):
             return self.create_response(500, unhandled_error.to_operation_outcome())
 
     def search_immunizations(self, aws_event) -> dict:
+        http_method = aws_event.get("httpMethod")
+        nhsNumberValue = None
+        diseaseTypeValue = None
+        nhsNumberParam = "-nhsNumber"
+        diseaseTypeParam = "-diseaseType"
+        if http_method == "POST":
+            # Check if the Content-Type is application/x-www-form-urlencoded
```

Comment:
Half of the comments in this section are not accurate to what is being done and the ones that are accurate are rather redundant. I think they can all be removed. Generally code should be self documenting and you'd only need comments for a rare instance of something unavoidably confusing.